### ISSUE TITLE
EKS - cannot edit EKS cluster on Rancher v2.2.0-rc2

### ISSUE THREAD

**Issue by jiaqiluo** [rancher/rancher#18391]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
- fresh install rancher v2.2.0-rc2
- add an EKS cluster
- edit the EKS cluster 
- change the value of min and max ASG value in order to scale up the cluster
![screenshot](https://user-images.githubusercontent.com/6218999/53389460-a466a980-394c-11e9-991e-2905ca15506b.png)

**Result:**
The state of the cluster goes to `updating` for a short time and then be `active` again, but the number of nodes is not changed

**Other details that may be helpful:**
- the PUT request is sent out and the response reflects the new values

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.2.0-rc2
- Installation option (single install/HA):
single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.5", GitCommit:"32ac1c9073b132b8ba18aa830f46b77dcceb0723", GitTreeState:"clean", BuildDate:"2018-06-21T11:46:00Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.5-eks-6bad6d", GitCommit:"6bad6d9c768dc0864dab48a11653aa53b5a47043", GitTreeState:"clean", BuildDate:"2018-12-06T23:13:14Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
(paste the output here)
```

**Comment by nathan-jenan-rancher** [comment]
Number of nodes is not a supported update for EKS clusters.  Only the K8s version can be updated currently.

**Comment by sangeethah** [comment]
@nathan-jenan-rancher In cases were updated are not supported , Ideally UI should not have presented the options ? Or when attempting to upgrade with unsupported fields, API should error out which gives clear indication to the user .

**Comment by jiaqiluo** [comment]
It is interesting that even though the UI shows the cluster is `updating` and has an error message, I can check the cluster, deploy workloads, run Kubectl, etc. 
<img width="1434" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/53437034-cf3b1700-39b9-11e9-943f-58cfa4c2429a.png">

**Comment by nathan-jenan-rancher** [comment]
I don't know what the desired behavior is.  Those all seem reasonable approaches to me though.

**Comment by cjellick** [comment]
Is this pre-existing then (like in 2.1.x) @nathan-jenan-rancher @sangeethah @jiaqiluo ?

**Comment by jiaqiluo** [comment]
This the screenshot of the page when editing an ESK cluster on Rancher v2.1.6.
We can change the value of min and max ASG value, but it takes no effect on AWS side. 
<img width="1253" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/53464194-9e350380-3a06-11e9-9f84-ac2ab466d7bc.png">

**Comment by nathan-jenan-rancher** [comment]
@cjellick EKS did not support _any_ updates until fairly recently (update was a no-op).  The PR that implemented update was only merged back in late Jan https://github.com/rancher/kontainer-engine/pull/109

**Comment by cjellick** [comment]
i really want to fix this, but it is pre-existing in 2.1.x and we dont have the bandwidth to address it in 2.2.0

**Comment by jiaqiluo** [comment]
The bug fix is validated on Rancher: master c098b38b

Confirmed that fields under the `Nodes` section are unchangeable in the UI when editing an EKS cluster (since they are not supported by AWS )
<img width="1365" alt="screenshot" src="https://user-images.githubusercontent.com/6218999/56603051-b616a700-65b3-11e9-8f2d-31379c98a687.png">
